bug(frontend): TerminalService.ts TS2352 — unsafe cast of Record<string,unknown>[] to SessionInfo[]

## Discovery Source
Found during implementation of #4537 — agent noted pre-existing error in TerminalService.ts.

## Problem
`autobot-frontend/src/services/TerminalService.ts:857` has:

```
error TS2352: Conversion of type 'Record<string, unknown>[]' to type 'SessionInfo[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```

A direct cast from `Record<string, unknown>[]` (likely parsed API response) to `SessionInfo[]` is rejected by TypeScript because the types don't share enough structure. This is an unsafe cast that bypasses type safety at the API boundary.

## Fix
Use a double cast through `unknown` to make intent explicit:
```typescript
(data as unknown as SessionInfo[])
```
Or better: use a type guard or `parseApiResponse<SessionInfo[]>()` to properly validate the shape at runtime.

## Files
- `autobot-frontend/src/services/TerminalService.ts` line 857

--- a/autobot-frontend/src/services/TerminalService.ts
+++ b/autobot-frontend/src/services/TerminalService.ts
@@ -854,7 +854,7 @@ class TerminalService {
   async getSessions(): Promise<SessionInfo[]> {
     try {
       const result = await apiClient.getTerminalSessions()
-      return (result || []) as SessionInfo[]
+      return (result || []) as unknown as SessionInfo[]
     } catch (error) {
       logger.error('Error getting sessions:', error)
       return []

